File: src/main/java/com/smarthome/server/entity/MinisterialEffect.java
package com.smarthome.server.entity;

import javax.persistence.*;

@Entity
public class MinisterialEffect {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String code;

    private String name;

    private String profile;

    private Integer stage;

    @OneToOne(cascade = CascadeType.ALL)
    private Area area;

    private String description;

    public Long getId() {
        return id;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getProfile() {
        return profile;
    }

    public void setProfile(String profile) {
        this.profile = profile;
    }

    public Integer getStage() {
        return stage;
    }

    public void setStage(Integer stage) {
        this.stage = stage;
    }

    public Area getArea() {
        return area;
    }

    public void setArea(Area area) {
        this.area = area;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: src/main/client/src/app/admin/ministerial-effect.ts
export class MinisterialEffect {

  private _id: number;
  private _code: string;
  private _name: string;
  private _profile: number;
  private _stage: number;
  private _area: number;
  private _description: string;

  get id(): number {
    return this._id;
  }

  get code(): string {
    return this._code;
  }

  get name(): string {
    return this._name;
  }

  get profile(): number {
    return this._profile;
  }

  get stage(): number {
    return this._stage;
  }

  get area(): number {
    return this._area;
  }

  get description(): string {
    return this._description;
  }

  set id(value: number) {
    this._id = value;
  }

  set code(value: string) {
    this._code = value;
  }

  set name(value: string) {
    this._name = value;
  }

  set profile(value: number) {
    this._profile = value;
  }

  set stage(value: number) {
    this._stage = value;
  }

  set area(value: number) {
    this._area = value;
  }

  set description(value: string) {
    this._description = value;
  }
}
